flux-middle/src/rty/fold.rs

```diff
@@ -195,6 +195,32 @@ pub trait TypeSuperVisitable: TypeVisitable {
     fn super_visit_with<V: TypeVisitor>(&self, visitor: &mut V) -> ControlFlow<V::BreakTy, ()>;
 }
 
+#[derive(Debug)]
+struct ProjectionTable(FxHashMap<AliasTy, Ty>);
```

Comment:
I would try to put stuff related to projection normalization in its own module as it's likely to become complicated quickly.